src/serac/physics/materials/thermal_expansion_material.hpp

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) 2019-2021, Lawrence Livermore National Security, LLC and
+// other Serac Project Developers. See the top-level LICENSE file for
+// details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+/**
+ * @file thermal_expansion_material.hpp
+ *
+ * @brief The thermal expansion material models for the solid module
+ */
+
+#pragma once
+
+#include "mfem.hpp"
+#include "serac/physics/utilities/finite_element_state.hpp"
+
+namespace serac {
+
+/**
+ * @brief Abstract interface class for a generic thermal expansion model
+ *
+ */
+class ThermalExpansionMaterial {
+public:
+  /**
+   * @brief Construct a new thermal expansion material
+   *
+   */
+  ThermalExpansionMaterial() : parent_to_reference_transformation_(nullptr) {}
+
+  /**
+   * @brief Destroy the thermal expansion material object
+   *
+   */
+  virtual ~ThermalExpansionMaterial() = default;
+
+  /**
+   * @brief Set the reference-to-target transformation. This is required to use coefficient parameters.
+   *
+   * @param[in] Ttr The reference-to-target (stress-free) transformation
+   */
+  void setTransformation(mfem::ElementTransformation& Ttr) { parent_to_reference_transformation_ = &Ttr; }
+
+  /**
+   * @brief Modify the displacement gradient to include thermal strain
+   *
+   * @param[inout] du_dX The unmodified displacement gradient
+   * @pre The underlying element transformation must be set to the correct integration point
+   * for the coefficient evaluations to work correctly
+   */
+  virtual void modifyDisplacementGradient(mfem::DenseMatrix& du_dX) = 0;
+
+protected:
+  /**
+   * @brief Non-owning pointer to the reference element to stree-free configuration (target) transformation
+   *
+   */
+  mfem::ElementTransformation* parent_to_reference_transformation_;
+};
+
+/**
+ * @brief A simple isotropic finite deformation thermal expansion module.
+ *
+ * @details This implements the split deformation gradient model
+ *
+ * \f[
+ * \mathbf{F} = \mathbf{F}_\theta \mathbf{F}_M
+ * \f]
+ *
+ * where \f$\mathbf{F}\f$ is the total deformation gradient, \f$\mathbf{F}_\theta\f$ is the
+ * deformation gradient induced by thermal expansion, and \f$\mathbf{F}_M\f$ is the mechanical
+ * deformation gradient. The thermal deformation gradient of this model is given by
+ *
+ * \f[
+ * \mathbf{F}_\theta = \mathbf{I} + \alpha \left( \theta - \theta_\textrm{ref} \right)
+ * \f]
```

Comment:
this renders to
![image](https://user-images.githubusercontent.com/61714427/134703644-2c051fbd-5238-46e9-8a4b-e9125dca5d30.png)
which doesn't make sense to me, do we want
```suggestion
 * \f[
 * \mathbf{F}_\theta = \alpha \left( \theta - \theta_\textrm{ref} \right) \mathbf{I}
 * \f]
```

![image](https://user-images.githubusercontent.com/61714427/134703865-5b32b1b7-691d-48ef-a89c-56fc1724ec22.png)

 instead?

Reply:
oops, my comment above is bad, probably this instead (?)
![image](https://user-images.githubusercontent.com/61714427/134704738-498693f4-fd16-4212-9459-a6fe8ff0b8b1.png)


```suggestion
 * \f[
 * \mathbf{F}_\theta = (1 + \alpha (\theta - \theta_\textrm{ref})) \mathbf{I}
 * \f]
```